pkg/agents/team7/agent1/agent.go

```diff
@@ -79,99 +93,539 @@ func New(baseAgent *infra.Base) (infra.Agent, error) {
 
-		if a.kindness < 0 {
-			a.kindness = 0
-		} else if a.kindness > 100 {
-			a.kindness = 100
+	if a.HP() >= tActive.ConditionValue() || int(a.CurrPlatFood()) >= tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
 		}
+	}
+	return foodtotake
+
+}
+func (a *CustomAgent7) treatyEQ(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
 
-		if a.greediness < 0 {
-			a.greediness = 0
-		} else if a.greediness > 100 {
-			a.greediness = 100
+	if a.HP() == tActive.ConditionValue() || int(a.CurrPlatFood()) == tActive.ConditionValue() || a.Floor() == tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
+			}
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
 		}
+	}
+	return foodtotake
 
-		foodtotake := food.FoodType(100 - a.kindness + a.greediness)
+}
+func (a *CustomAgent7) treatyLT(tActive messages.Treaty, foodtotake food.FoodType, available food.FoodType, amount food.FoodType) food.FoodType {
 
-		//has the agent seen the platform
-		if a.CurrPlatFood() != -1 && !a.seenPlatform {
-			foodEaten, err := a.TakeFood(foodtotake)
-			if err != nil {
-				switch err.(type) {
-				case *infra.FloorError:
-				case *infra.NegFoodError:
-				case *infra.AlreadyEatenError:
-				default:
-				}
+	if a.Floor() < tActive.ConditionValue() {
+		switch tActive.RequestOp() {
+		case messages.GT:
+			if foodtotake > available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount) - 1
 			}
-			a.foodEaten = foodEaten
-			if a.foodEaten > 0 {
-				a.daysHungry = 0
-			} else {
-				a.daysHungry++
+		case messages.GE:
+			if foodtotake >= available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
+			}
+		case messages.EQ:
+			if foodtotake != available-food.FoodType(tActive.RequestValue()) {
+				foodtotake = available - food.FoodType(amount)
 			}
-			a.Log("Team 7 has seen the platform:", infra.Fields{"foodEaten": a.foodEaten, "health": a.HP(), "daysHungry": a.daysHungry})
-			a.seenPlatform = true
 		}
```

Comment:
You've copied this same block of code five times - you can put this into a function.